pim-tool-back/src/main/java/vn/elca/training/service/impl/ProjectServiceImpl.java

```diff
@@ -1,93 +1,193 @@
 package vn.elca.training.service.impl;
 
+import com.querydsl.core.types.dsl.BooleanExpression;
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Primary;
-import org.springframework.context.annotation.Profile;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
-
-import vn.elca.training.exception.UpdateProjectException;
 import vn.elca.training.model.dto.ProjectDto;
 import vn.elca.training.model.entity.Project;
+import vn.elca.training.model.entity.QProject;
+import vn.elca.training.model.exception.*;
+import vn.elca.training.repository.EmployeeRepository;
+import vn.elca.training.repository.GroupRepository;
 import vn.elca.training.repository.ProjectRepository;
 import vn.elca.training.service.ProjectService;
+import vn.elca.training.util.ApplicationMapper;
+import vn.elca.training.validator.ProjectValidator;
 
-import java.time.LocalDate;
+import java.util.HashSet;
 import java.util.List;
 
-/**
- * @author vlp
- * 
- */
 @Service
-@Profile("!dummy | dev")
-@Primary
+@Transactional
 public class ProjectServiceImpl implements ProjectService {
 
     @Autowired
     private ProjectRepository projectRepository;
 
-    @Override
-    public List<Project> findAll() {
-        return projectRepository.findAll();
-    }
+    @Autowired
+    private EmployeeRepository employeeRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Autowired
+    private ApplicationMapper mapper;
+
+    @Autowired
+    private ProjectValidator projectValidator;
 
     @Override
-    public long count() {
-        return projectRepository.count();
+    public ProjectDto create(ProjectDto projectDto) throws StartDateAfterEndDateException,
+            ProjectNumberAlreadyExistsException, InvalidProjectMembersException {
+        // Validate the input data before creating the project
```

Comment:
[N} No need to comment every line of code, especially when the method names are clear enough